[BE] Preparing environment for tests

MihaiDruta opened the pull request:
[BE] Preparing environment for tests
https://trello.com/c/Hz9GcyOy/93-be-preparing-environment-for-tests

1.Installed LiipTestFixturesBundle and LiipFunctionalTestBundle
2.Configured new SQLite database for tests
3.First functional test on successful registering

Commit:
[BE] Preparing environment for tests
https://trello.com/c/Hz9GcyOy/93-be-preparing-environment-for-tests

1.Configured new SQLite database for tests
2.First functional test on successful registering

SpartakusMd commented on config/packages/test/framework.yaml:
```diff
@@ -2,3 +2,5 @@ framework:
   session:
     storage_id: session.storage.mock_file
+
+liip_functional_test: ~
```
Move this to a separate yaml file.

SpartakusMd commented on tests/RegisterTest.php:
```diff
@@ -0,0 +1,38 @@
+            json_encode($data)
+        );
+
+        $this->assertEquals(200, $client->getResponse()->getStatusCode());
```
Check that the data in the database is present and is correct.

Commit:
[BE] Preparing environment for tests
https://trello.com/c/Hz9GcyOy/93-be-preparing-environment-for-tests

1.Created new liip_functional_test.yaml config file
2.Verified if the data is in the database and is correct after successful registration

SpartakusMd commented on tests/RegisterTest.php:
```diff
@@ -34,5 +39,32 @@ public function testSuccessfullRegister()
+        /** @var User $user */
+        $user = $repository->findOneBy(array(
+            'username' => 'TestUsername',
+            'email' => 'testemail@test.mail',
```
Search only by username as it is unique.

SpartakusMd commented on tests/RegisterTest.php:
```diff
@@ -0,0 +1,70 @@
+
+        $repository = $this->entityManager->getRepository(User::class);
+
+        $users = $repository->findBy(array(
```
Why 2 queries? Use only the second one with only username as search parameter.

Commit:
[BE] Preparing environment for tests
https://trello.com/c/Hz9GcyOy/93-be-preparing-environment-for-tests

Removed unnecessary query from test and searching only by username in the remaining one.
